### PR TITLE
refactor: config options combined into DNS proxy config struct

### PR DESCRIPTION
**Pull request by vipul-21**
Config options combined into a single struct `DNSProxyConfig` instead of passing individual config options.

### CODE REVIEW

File: pkg/fqdn/dnsproxy/proxy.go [hunk 1]
```diff
@@ -615,6 +615,18 @@ func (proxyStat *ProxyRequestContext) IsTimeout() bool {
 	return false
 }
 
+// DNSProxyConfig struct
```
Comment (tklauser) [inline pkg/fqdn/dnsproxy/proxy.go]: nit: let's a a slightly more verbose godoc comment here, e.g.

```suggestion
// DNSProxyConfig is the configuration for the DNS proxy.
```

**Review by tklauser — APPROVED** [review]
LGTM with one small nit about `DNSProxyConfig` godoc.

**Review by pippolo84 — APPROVED** [review]
LGTM (minus Tobias suggestion) :+1:

**Comment by pippolo84** [comment]
Removing "ready-to-merge" label until Tobias comment is addressed.

**Review by pippolo84 — APPROVED** [review]
Thanks

**Pull request merged**